src/DynamoCore/Models/RecordableCommands.cs

```diff
@@ -689,6 +689,110 @@ protected override void SerializeCore(XmlElement element)
             #endregion
         }
 
+        [DataContract]
+        public class AutoCreateNodeCommand : ModelBasedRecordableCommand
+        {
+            #region Public Class Methods
+
+            private void SetProperties(int outputPortIndex, int inputPortIndex, double x, double y, 
+                bool defaultPosition, bool transformCoordinates)
+            {
+                OutputPortIndex = outputPortIndex;
+                InputPortIndex = inputPortIndex;
+                X = x;
+                Y = y;
+                DefaultPosition = defaultPosition;
+                TransformCoordinates = transformCoordinates;
+            }
+
+            public AutoCreateNodeCommand(
+                NodeModel inputNode, NodeModel givenNode, int outputPortIndex, int inputPortIndex, 
+                double x, double y, bool defaultPosition, bool transformCoordinates)
+                : base(inputNode != null && givenNode != null ? new[] { inputNode.GUID, givenNode.GUID } : new[] { Guid.Empty })
+            {
+                InputNode = inputNode;
+                GivenNode = givenNode;
+                
+                SetProperties(outputPortIndex, inputPortIndex, x, y, defaultPosition, transformCoordinates);
+            }
+
+            #endregion
+
+            #region Public Command Properties
```

Comment:
Let's see how many of these properties are not necessary, some of them might already be exposed from base class? These properties are meant to be used in `DynamoModel.CreateAndConnectNodeImpl` as outlined below.